File: slp/backend-dotnet/Features/Auth/AuthDTO.cs
namespace backend_dotnet.Features.Auth
{
    public record LoginRequest(string Username, string Password);

    public record ForgotPasswordRequest(string Email);

    public record ResetPasswordRequest(string Token, string NewPassword);

    public record VerifyEmailRequest(string Token);

    public record UpdateUserRequest(string Name, string AvatarUrl);

    public record ChangePasswordRequest(string CurrentPassword, string NewPassword);

    public class LoginResponse
    {
        public string Token { get; set; } = "";
        public string UserId { get; set; } = "";
        public string Email { get; set; } = "";
    }

    public record RegisterUserRequest(
        string Username,
        string Email,
        string Password
    );

    public class LoginResult
    {
        public bool Success { get; set; }
        public LoginResponse? Data { get; set; }
        public string? ErrorCode { get; set; }   // e.g., "USER_NOT_FOUND", "INVALID_PASSWORD", "ACCOUNT_BANNED", "EMAIL_NOT_VERIFIED"
        public string? Message { get; set; }
    }

    public class CurrentUserDto
    {
        public int Id { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public bool EmailConfirmed { get; set; }
        public string Role { get; set; }
        public string Status { get; set; }
        public string? AvatarFilename { get; set; }
        public DateTime CreatedAt { get; set; }
    }

    public class ChangePasswordResult
    {
        public bool Success { get; set; }
        public string? ErrorCode { get; set; }  // "USER_NOT_FOUND" | "INVALID_CURRENT_PASSWORD"
        public string? Message { get; set; }
    }
}

File: slp/backend-java/src/main/java/com/app/backendjava/features/auth/AuthDto.java
package com.app.backendjava.features.auth;

import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;

public class AuthDto {
    public record LoginRequest(String username, String password) {}
    public record ForgotPasswordRequest(String email) {}
    public record ResetPasswordRequest(String token, String newPassword) {}
    public record VerifyEmailRequest(String token) {}
    public record UpdateUserRequest(String name, String avatarUrl) {}
    public record ChangePasswordRequest(String currentPassword, String newPassword) {}
    public record RegisterUserRequest(String username, String email, String password) {}

    @Data
    public static class LoginResponse {
        private String token = "";
        private String userId = "";
        private String email = "";
    }

    @Data
    @Builder
    public static class LoginResult {
        private boolean success;
        private LoginResponse data;
        private String errorCode;
        private String message;
    }

    @Data
    public static class CurrentUserDto {
        private Integer id;
        private String username;
        private String email;
        private boolean emailConfirmed;
        private String role;
        private String status;
        private String avatarFilename;
        private LocalDateTime createdAt;
    }

    @Data
    @Builder
    public static class ChangePasswordResult {
        private boolean success;
        private String errorCode;
        private String message;
    }
}